src/test/java/net/snowflake/client/jdbc/ResultSetJsonVsArrowIT.java

```diff
@@ -1497,21 +1484,21 @@ public void TestArrowStringRoundTrip() throws SQLException {
     String big_number = "11111111112222222222333333333344444444";
     try (Connection con = init()) {
       Statement st = con.createStatement();
+      st.execute("alter session set AUTOCOMMIT=FALSE");
       for (int i = 0; i < 38; i++) {
         StringBuilder to_insert = new StringBuilder(big_number);
         if (i != 0) {
           int insert_to = 38 - i;
           to_insert.insert(insert_to, ".");
         }
         st.execute("create or replace table test_arrow_string (a NUMBER(38, " + i + ") )");
-        st.execute("begin");
```

Comment:
Taking out transactions did not stop the check from hanging, but setting autocommit to false works instead. This should have the same effect as rolling back the transaction.

Reply:
I missed this comment. I still don't get why AUTOCOMMIT changes the behavior.
By default, AUTOCOMMIT is true, and BEGIN, COMMIT, ROLLBACK won't take any effect in theory, though, removing them will at least any impact of those commands if any.